src/extblkdev/ExtBlkDevPlugin.cc

```diff
@@ -0,0 +1,218 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*- 
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph distributed storage system
+ *
+ * (C) Copyright IBM Corporation 2022
+ * Author: Martin Ohmacht <mohmacht@us.ibm.com>
+ *
+ * Based on the file ceph/src/erasure-code/ErasureCodePlugin.cc
+ * Copyright (C) 2013,2014 Cloudwatt <libre.licensing@cloudwatt.com>
+ * Copyright (C) 2014 Red Hat <contact@redhat.com>
+ *
+ * Author: Loic Dachary <loic@dachary.org>
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ * 
+ */
+
+#include <errno.h>
+
+#include "ceph_ver.h"
+#include "ExtBlkDevPlugin.h"
+#include "common/errno.h"
+#include "include/dlfcn_compat.h"
+#include "include/str_list.h"
+#include "include/ceph_assert.h"
+#include "common/ceph_context.h"
+#include "common/debug.h"
+
+#define dout_subsys ceph_subsys_context
+
+using namespace std;
+
+namespace ceph {
+
+  namespace extblkdev {
+
+
+    // iterate across plugins and determine each capability reqirement
+    // and trim permitted set accordingly
+    int limit_caps(CephContext *cct)
+    {
+      int rc=0;
```

Comment:
General comment: please adhere more to code formatting style:
https://github.com/ceph/ceph/blob/main/CodingStyle

In this case I have in mind spaces " " around operators:
https://google.github.io/styleguide/cppguide.html#Horizontal_Whitespace

Reply:
Fixed whitespaces across all files. Are there other major violations that you observed?